file.go

```diff
@@ -0,0 +1,289 @@
+package ipam
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"sync"
+	"time"
+)
+
+type file struct {
+	path       string
+	prettyJSON bool
+	modTime    time.Time
+	parent     Storage
+	lock       sync.RWMutex
+}
+
+var (
+	nullModTime          time.Time
+	DefaultLocalFilePath string
+)
+
+type fileJSONData map[string]map[string]prefixJSON
+
+func init() {
+	nullModTime = time.Unix(0, 0)
+	DefaultLocalFilePath = path.Join(getXDGDataHome(), "go-ipam", "ipam-db.json")
+}
+
+func getXDGDataHome() string {
+	if val := os.Getenv("XDG_DATA_HOME"); val != "" {
+		return val
+	}
+
+	val, err := os.UserHomeDir()
+	if err != nil {
+		val = "."
+	} else {
+		val = path.Join(val, ".local", "share")
+	}
+	return val
+}
+
+// NewLocalFile creates a JSON file storage for ipam
+func NewLocalFile(ctx context.Context, path string) Storage {
+	return &file{
+		path:       path,
+		prettyJSON: true,
+		parent:     NewMemory(ctx),
+		modTime:    nullModTime,
+		lock:       sync.RWMutex{},
+	}
+}
+
+func (f *file) clearParent(ctx context.Context) (err error) {
+	namespaces, err := f.parent.ListNamespaces(ctx)
+	if err != nil {
+		return err
+	}
+	for _, namespace := range namespaces {
+		if err = f.parent.DeleteAllPrefixes(ctx, namespace); err != nil {
+			return err
+		}
+		if namespace == defaultNamespace {
+			// skip deletion instead of replicating NewMemory behavior
+			continue
+		}
+		if err = f.parent.DeleteNamespace(ctx, namespace); err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func (f *file) refresh(ctx context.Context) error {
+	if modTime := f.getModTime(); modTime != nullModTime && modTime == f.modTime {
+		return nil
+	}
+	return f.reload(ctx)
+}
+func (f *file) getModTime() time.Time {
+	info, err := os.Stat(f.path)
+	if err != nil {
+		return nullModTime
+	}
+	return info.ModTime()
+}
+
+// see ipamer.NamespacedLoad for similar, but incomplete functionality
+func (f *file) reload(ctx context.Context) (err error) {
```

Comment:
A bit more documentation would be nice here, because its hard to understand what is the difference between reload and refresh.